### DISCUSSION TITLE
seq! with partial byte fields?

### DISCUSSION THREAD

**Discussion by ckoehler** [winnow-rs/winnow#539] [Q&A] [answered]
Hi there!

What's the best way to parse something like `0b1000_0100`, where `0b10.._....` is one field, and `0b..00_0100` is another? This is in the midst of other fields that are on byte boundaries.

I could just parse that byte and then refer back to it, but that would require defining it in the struct:

```rust
struct Packet {
  the_byte: u8,
  field1: u8,
  field2: u8,
}
```

and then in the parse function something like (untested):

```rust
seq! {Self{
    the_byte: be_u8,
    field1: (the_byte & 0xc0) >> 6,
    field2: the_byte & 0x3f,
}}.parse_next(i)
```

Maybe that's the best way, just wanted to throw this out.

Thanks! 

Christoph

**Comment by epage** [comment, marked as answer]
That or not use `seq!` and maybe even parse more imperatively.

**Reply by epage** [reply]
You could also make `field1` and `field2` accessors, rather than direct fields

**Reply by ckoehler** [reply]
True. Thanks for the confirmation!
